src/vm/object.cpp

```diff
@@ -1525,21 +1525,21 @@ void STDCALL CopyValueClassUnchecked(void* dest, void* src, MethodTable *pMT)
     switch (pMT->GetNumInstanceFieldBytes())
     {        
     case 1:
-        VolatileStore((UINT8*)dest, *(UINT8*)src);
+        *(UINT8*)dest = *(UINT8*)src;
         break;
 #ifndef ALIGN_ACCESS
         // we can hit an alignment fault if the value type has multiple 
         // smaller fields.  Example: if there are two I4 fields, the 
         // value class can be aligned to 4-byte boundaries, yet the 
         // NumInstanceFieldBytes is 8
```

Comment:
> The default case does not use. The 2,4,8 cases only exist if ALIGN_ACCESS is not defined.

This looks broken.

> I can add an optional isUnaligned parameter to VolatileStore...

I think it may be better to add optional flag to CopyValueClass that let's you supress the barrier in it. Then pass this flag from places like boxing where the barrier is not needed.